core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -360,9 +360,10 @@ protected void doSample() {
                     try {
                         // its a listed node, light connect to it...
                         logger.trace("connecting to listed node (light) [{}]", listedNode);
-                        transportService.connectToNodeLight(listedNode);
+                        transportService.connectToNodeLight(listedNode, pingTimeout, !ignoreClusterName);
```

Comment:
> Though I wonder if we need a new action - why not just use the liveness action?

The only argument that I have is that I think that the handshake action should be exposed in the transport tracer but do not think that the liveness action should. If these were folded into one action, that would not be possible. If you think that's a poor argument, I'll investigate refactoring into one action.